pkg/virt-controller/watch/vm.go

```diff
@@ -175,18 +177,25 @@ func (c *VMController) needsSync(key string) bool {
 	return c.expectations.SatisfiedExpectations(key) && c.dataVolumeExpectations.SatisfiedExpectations(key)
 }
 
+var virtControllerTracerVM *traceUtils.Tracer
+
 func (c *VMController) Execute() bool {
 	key, quit := c.Queue.Get()
 	if quit {
 		return false
 	}
+
+	virtControllerTracerVM = traceUtils.NewTrace(time.Second, "virt-controller VM workqueue", trace.Field{Key: "Workqueue Key", Value: key})
+
 	defer c.Queue.Done(key)
 	if err := c.execute(key.(string)); err != nil {
 		log.Log.Reason(err).Infof("re-enqueuing VirtualMachine %v", key)
 		c.Queue.AddRateLimited(key)
+		virtControllerTracerVM.Trace.Step("RateLimit VM Key", trace.Field{Key: "key", Value: key})
```

Comment:
I think this needs to be a ```LogIfLong``` call instead of a ```Step```

Like i point out in another comment, I think we should just use a ```defer``` statement for LogIfLong in these execute functions.